clients/src/main/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManager.java

```diff
@@ -215,6 +216,7 @@ public NetworkClientDelegate.PollResult poll(long currentTimeMs) {
         }
 
         NetworkClientDelegate.UnsentRequest request = makeHeartbeatRequest(currentTimeMs, false);
+        membershipManager.onHeartbeatRequestSent();
```

Comment:
yes, makes sense, done. I had considered it but wasn't convinced thinking that it would complicate the leaving on the membershipMgr, but with your comment I took another round and it does seem better actually, just a single leaving, that in some cases is due to poll timer expired (and that allows the manager to make the right transitions after leaving).